### DISCUSSION TITLE
Hardlinking into Dokan?

### DISCUSSION THREAD

**Discussion by jbayardo** [dokan-dev/dokany#980] [Q&A]
Is there any way to set Dokan up to enable hardlinking into the Dokan folder? For example, if my virtualization root is C:\dokan, and there's a virtual file in C:\dokan\a.txt, I'd like to be able to hardlink from C:\a.txt to C:\dokan\a.txt. 

This is required in my use-case due to some obscure requirements from the tools that will access C:\a.txt (namely, that they are acutely aware of symlinks, and so having a symlink instead would break things).

Currently, if I try to do this I get an ERROR_NOT_SAME_DEVICE (which makes sense). Is there any way one can make Windows believe that the Dokan mount is "the same device" as C:?

**Comment by Liryna** [comment]
Hi @jbayardo ,

Hardlink is just not supported in Dokan. Regarding your used case, I am even not sure if Dokan would even be query for the link creation or it would just fail way before when contacting `C` drive.
Would need to enable the driver logs and see if there is any activity to confirm this point.
